frontend/templates/troubleshooting/server.tsx

```diff
@@ -53,25 +54,33 @@ export const getServerSideProps: GetServerSideProps<
                      ifixitOrigin,
                   },
                   { forceMiss: hasDisableCacheGets(context) }
-               )
-         )
+               ).catch(() => null)
+            );
+         })
       );
       return {
          ...solution,
-         guides,
+         guides: guides.filter((guide): guide is Guide => Boolean(guide)),
          products: products.filter((product): product is ProductType =>
             Boolean(product)
          ),
       };
    }
 
-   const troubleshootingData = await client.get<TroubleshootingApiData>(
-      `Troubleshooting/${wikiname}`,
-      'troubleshooting',
-      {
-         credentials: 'include',
-      }
-   );
+   let troubleshootingData: TroubleshootingApiData;
+   try {
+      troubleshootingData = await client.get<TroubleshootingApiData>(
+         `Troubleshooting/${wikiname}`,
+         'troubleshooting',
+         {
+            credentials: 'include',
+         }
+      );
+   } catch (e) {
```

Comment:
I think it's a bad idea here to swallow all exceptions. Do 404s throw exceptions? Can we narrow this handling to *just* deal with 404s?

Reply:
The exception comes from here: https://github.com/iFixit/react-commerce/blob/0ff5ad629f7070ce81339a87e920a597d578c77d/packages/ifixit-api-client/index.tsx#L80-L82
We could check against the response's status text to serve a different error page. Maybe just re-throw the exception which should get us the error page: https://github.com/iFixit/react-commerce/blob/c55d58bf0b0f150baff1486b85738a773c6cc6c0/frontend/pages/_error.tsx

Reply:
Re-throwing the exception should also show the error page for the exceptions that `fetch()` can throw. https://developer.mozilla.org/en-US/docs/Web/API/fetch#exceptions

Reply:
[`f6e86bc` (#1705)](https://github.com/iFixit/react-commerce/pull/1705/commits/f6e86bc1b690baa9c39121f38c6cbf57ad030429)